https://bugs.gentoo.org/506498

From d4ccb5e05c99c4006fe43ab08ebe13b7a74fc111 Mon Sep 17 00:00:00 2001
From: Tristan Gingold <tristan.gingold@adacore.com>
Date: Mon, 17 Mar 2014 14:01:02 +0100
Subject: [PATCH] darwin: handle recent version of dyld

gdb/
	* solib-darwin.c (DYLD_VERSION_MAX): Increase value.
	(darwin_solib_create_inferior_hook): Emit a warning if version
	is unhandled.
---
 gdb/ChangeLog      | 6 ++++++
 gdb/solib-darwin.c | 7 +++++--
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/gdb/solib-darwin.c b/gdb/solib-darwin.c
index e8d4667..a9989ea 100644
--- a/gdb/solib-darwin.c
+++ b/gdb/solib-darwin.c
@@ -70,7 +70,7 @@ struct gdb_dyld_all_image_infos
 
 /* Current all_image_infos version.  */
 #define DYLD_VERSION_MIN 1
-#define DYLD_VERSION_MAX 12
+#define DYLD_VERSION_MAX 14
 
 /* Per PSPACE specific data.  */
 struct darwin_info
@@ -513,7 +513,10 @@ darwin_solib_create_inferior_hook (int from_tty)
   darwin_load_image_infos (info);
 
   if (!darwin_dyld_version_ok (info))
-    return;
+    {
+      warning (_("unhandled dyld version (%d)"), info->all_image.version);
+      return;
+    }
 
   create_solib_event_breakpoint (target_gdbarch (), info->all_image.notifier);
 
-- 
1.9.2

